mm2src/coins/eth.rs

```diff
@@ -1317,10 +1319,15 @@ impl SwapOps for EthCoin {
     #[inline]
     fn derive_htlc_pubkey(&self, _swap_unique_data: &[u8]) -> Vec<u8> {
         match self.priv_key_policy {
-            EthPrivKeyPolicy::KeyPair(ref key_pair) => key_pair_from_secret(key_pair.secret().as_bytes())
+            EthPrivKeyPolicy::Iguana(ref key_pair)
+            | EthPrivKeyPolicy::HDWallet {
+                activated_key: ref key_pair,
+                ..
+            } => key_pair_from_secret(key_pair.secret().as_bytes())
                 .expect("valid key")
                 .public_slice()
                 .to_vec(),
+            EthPrivKeyPolicy::Trezor => todo!(),
```

Comment:
[the same](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1933/files#r1289630217)

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/1933#discussion_r1294099638